cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -458,19 +464,21 @@ public void resumeLoadMonitorActivity() {
 
   /**
    * Optimize a cluster workload model.
-   * @param goals a list of goals to optimize. When empty all goals will be used.
-   * @param requirements the model completeness requirements to enforce when generating the proposals.
-   * @param operationProgress the progress of the job to report.
+   * @param goals A list of goals to optimize. When empty all goals will be used.
+   * @param requirements The model completeness requirements to enforce when generating the proposals.
+   * @param operationProgress The progress of the job to report.
    * @param allowCapacityEstimation Allow capacity estimation in cluster model if the requested broker capacity is unavailable.
+   * @param skipHardGoalCheck Whether check and add all hard goals to {@code goals}.
```

Comment:
If we rephrase it as follows, would it be more clear?
`Whether check and add all hard goals to {@code goals}.` -> `True if the provided {@code goals} do not have to contain all hard goals, false otherwise`